mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h

```diff
@@ -4012,12 +4023,13 @@ struct ThreadwiseCopyRewritePattern
     // Determine if we need to emit codes for out-of-bound check.
     bool toEmitOOBCheckLogic = false;
     SmallVector<unsigned, 2> oobCheckDims;
-    if (sourceTransform) {
-      for (unsigned iter = 0; iter < sourceTransform.getNumResults(); ++iter) {
-        auto expr = sourceTransform.getResult(iter);
-        if (hasPadding(expr)) {
-          toEmitOOBCheckLogic = true;
+    if (limitDimTransform) {
+      auto limits = limitDimTransform.getResults();
+      for (unsigned iter = 0; iter < limits.size(); ++iter) {
+        auto expr = limits[iter];
+        if (isOOBCheck(expr)) {
```

Comment:
it's better  that both  isOOBCheck and  hasPadding execute

Reply:
on the contrary, remove `hasPadding` once this PR gets merged.